app/models/grda_warehouse/warehouse_reports/project/data_quality/version_three.rb

```diff
@@ -0,0 +1,1988 @@
+        refused_income_at_exit_percentage = 0 # (refused_income_at_exit.size.to_f/exit_event_count*100).round(2) rescue 0
+      end
+
+      # missing and refused destinations will be NaN if there are no leavers
+      if leavers.count == 0
+        missing_destination_percentage = 0
+        refused_destination_percentage = 0
+      else
+        missing_destination_percentage = (missing_destination.size.to_f/leavers_count*100).round(2) rescue 0
+        refused_destination_percentage = (refused_destination.size.to_f/leavers_count*100).round(2) rescue 0
+      end
+      answers = {
+        missing_disabling_condition: missing_disabling_condition.size,
+        missing_disabling_condition_percentage: missing_disabling_condition_percentage,
+        missing_prior_living_situation: missing_prior_living.size,
+        missing_prior_living_situation_percentage: missing_prior_living_percentage,
+        missing_destination: missing_destination.size,
+        missing_destination_percentage: missing_destination_percentage,
+        refused_disabling_condition: refused_disabling_condition.size,
+        refused_disabling_condition_percentage: refused_disabling_condition_percentage,
+        refused_prior_living_situation: refused_prior_living.size,
+        refused_prior_living_situation_percentage: refused_prior_living_percentage,
+        refused_destination: refused_destination.size,
+        refused_destination_percentage: refused_destination_percentage,
+        missing_income_at_entry: missing_income_at_entry.size,
+        refused_income_at_entry: refused_income_at_entry.size,
+        missing_income_at_exit: missing_income_at_exit.size,
+        refused_income_at_exit: refused_income_at_exit.size,
+        missing_income_at_entry_percentage: missing_income_at_entry_percentage,
+        refused_income_at_entry_percentage: refused_income_at_entry_percentage,
+        missing_income_at_exit_percentage: missing_income_at_exit_percentage,
+        refused_income_at_exit_percentage: refused_income_at_exit_percentage,
+      }
+
+      support = {
+        missing_disabling_condition: {
+          headers: ['Client ID'],
+          counts: missing_disabling_condition.map{|m| Array.wrap(m)}
+        },
+        missing_prior_living_situation: {
+          headers: ['Client ID'],
+          counts: missing_prior_living.map{|m| Array.wrap(m)}
+        },
+        missing_destination: {
+          headers: ['Client ID'],
+          counts: missing_destination.map{|m| Array.wrap(m)}
+        },
+        refused_disabling_condition: {
+          headers: ['Client ID'],
+          counts: refused_disabling_condition.map{|m| Array.wrap(m)}
+        },
+        refused_prior_living_situation: {
+          headers: ['Client ID'],
+          counts: refused_prior_living.map{|m| Array.wrap(m)}
+        },
+        refused_destination: {
+          headers: ['Client ID'],
+          counts: refused_destination.m
```

Comment:
This may need to be a bit smarter.  I think it's possible to have the income assessment, but have answered in such a way as to indicate that it is missing.

Reply:
Also, let's name these arguments, at least the second one.

Reply:
This needs to check for missing records and TotalMontlyIncome == null